docs/source/DeveloperGuide/Python3Migration/index.rst

```diff
@@ -0,0 +1,255 @@
+
+  These three are all confined to only affecting the current file making it easy to progressively add them to individual files without unexpected side effects in other parts of DIRAC.
+  While ``from __future__ import unicode_literals`` also exists, this tends to result in unexpected side effects from ``unicode`` objects being passed to functions that weren't designed to handle them and as a result it is not expected to be useful for DIRAC's Python 3 migration.
+
+**bytes vs str**
+  The most difficult change when moving Python 3 is the splitting of the `str` type one for text and one for true binary data.
+  This exposes subtle issues in Python 2 that were likely never noticed and an automatic conversion to fix this is inherently impossible.
+  More details about this can be found `here <https://portingguide.readthedocs.io/en/latest/strings.html>`_.
+
+  In most situations DIRAC is only dealing with ascii or unicode strings and therefore nothing needs to change.
+  However many libraries choose to be independent of the character encoding used and therefore return a ``bytes`` object in Python 3 instead of ``str``
+
+  .. code-block:: python
+
+    result = subprocess.check_output(["echo", "Hello"])
+
+    # Bad: Fails on Python 3 with "TypeError: can't concat str to bytes"
+    return "Result is" + result
+
+    # Good: Explicitly decode bytes to str (does nothing on Python 2)
+    return "Result is" + result.decode()
+
+    # For subprocess functions, the universal_newlines=True argument can be used
+    other_result = subprocess.check_output(["echo", "Hello"], universal_newlines=True)
+    # Good: other_result is already a str object
+    return "Result is" + other_result
+
+  **Checking the type of a string:**
+
+  .. code-block:: python
+
+    # Bad: Types should be check using isinstance
+    if type(my_variable) == str:
+
+    # Bad: basestring does not exist in Python 3
+    if isinstance(my_variable, basestring):
+
+    # Good: Supports both Python 2 and 3
+    if isinstance(my_variable, six.string_types):
+
+  **Reading files:**
+
+  It's preferable to explicitly state if a file is being opened in text mode or binary mode.
+
+  .. code-block:: python
+
+    # Bad: Works but it is unclear if data is expected to bytes or a string
+    with open("my_file.txt") as fp:
+       data = fp.read().split("\n")
+
+    # Good: File is explicitly in text mode
+    with open("my_file.txt", "rt") as fp:
+       data = fp.read().split("\n")
+
+    # Bad: Fails on python 3 as "\n" is a string not bytes
+    with open("my_file.txt", "rb") as fp:
+       data = fp.read().split("\n")
+
+    # Good: Prefix the "\n" to make it a bytes object
+    with open("my_file.txt", "rb") as fp:
+       data = fp.read().split(b"\n")
```

Comment:
What about a recommendation that one should rather use `io.open` instead of the built-in `open`?